bucket.go

```diff
@@ -291,6 +291,59 @@ func (b *Bucket) DeleteBucket(key []byte) error {
 	return nil
 }
 
+// MoveBucket moves a sub-bucket from the source bucket to the destination bucket.
+// Returns an error if
+//  1. the sub-bucket cannot be found in the source bucket;
+//  2. or the key already exists in the destination bucket;
+//  3. the key represents a non-bucket value.
+func (b *Bucket) MoveBucket(key []byte, dstBucket *Bucket) error {
+	if b.tx.db == nil || dstBucket.tx.db == nil {
+		return errors.ErrTxClosed
+	} else if !dstBucket.Writable() {
+		return errors.ErrTxNotWritable
+	}
+
+	// Move cursor to correct position.
+	c := b.Cursor()
+	k, v, flags := c.seek(key)
+
+	// Return an error if bucket doesn't exist or is not a bucket.
+	if !bytes.Equal(key, k) {
+		return errors.ErrBucketNotFound
+	} else if (flags & common.BucketLeafFlag) == 0 {
+		return fmt.Errorf("key %q isn't a bucket in the source bucket: %w", key, errors.ErrIncompatibleValue)
+	}
+
+	// Do nothing (return true directly) if the source bucket and the
+	// destination bucket are actually the same bucket.
+	if b == dstBucket || (b.RootPage() == dstBucket.RootPage() && b.RootPage() != 0) {
```

Comment:
`b.RootPage() != 0`

 iiuc, this means the rootPage can not be `inlineBucket` or ?

Reply:
> this means the rootPage can not be `inlineBucket` or ?

Right. It means we don't check inline buckets.

Reply:
but this mean that if the srcBucket == dstBucket, but the bucket is `inlined` , then we still move the bucket 

i think the current logic will move the inLinedBucket into childBucket, please correct me if i am wrong

Reply:
An inline bucket can't contain child buckets. In other words, if a bucket contains child buckets, then it isn't an inline bucket for sure.

Reply:
I am thinking probably it makes more sense to return an error (e.g. "`the source and target buckets are the same bucket`") in such case.

Similar to
```
# mv aa aa
mv: 'aa' and 'aa' are the same file
```

Reply:
+1 i agree 👍🏽

Reply:
done